packages/cdk/utils/src/replace-ranges-with-seconds/replace-with-numbers.ts

```diff
@@ -0,0 +1,64 @@
+import { Nullable } from '@angular-ru/cdk/typings';
+
+import { isNil } from '../is-nil';
+import { isNotNil } from '../is-not-nil';
+import { ReplaceNumber } from './types';
+
+const REGEXP_NUMBER: string = `[0-9]+([\.][0-9]+)?`;
+
+export function replaceWithNumbers(value: string, replaceMapList: ReplaceNumber[]): string {
+    const validPostfixList: string = getValidPostfixList(replaceMapList);
+    const matchPattern: RegExp = new RegExp(`(^|\\s)(${REGEXP_NUMBER})(${validPostfixList})(\\s|$)`);
+    let result: string = value;
+    let match: Nullable<string> = null;
+
+    while (isNotNil((match = matchPattern.exec(result)?.[0]))) {
```

Comment:
can we simplify `while ( hasMatcher)  {`?

Reply:
good, I can simply use test()